### ISSUE TITLE
[com_fields] Radio custom field on User fails with DUPLICATE KEY

### ISSUE THREAD

**Issue by solracsf** [joomla/joomla-cms#26153]
### Steps to reproduce the issue

Create a User > Radio custom field, editable in both Front an Admin. 

![image](https://user-images.githubusercontent.com/12234510/64176913-a90d9c80-ce5e-11e9-9f53-ced9c9ec77c2.png)

![image](https://user-images.githubusercontent.com/12234510/64176959-c5113e00-ce5e-11e9-9edd-fe14f4383ef3.png)

### Expected result

User can switch the radio (**On** or **Off**) on their profile or by an admin.

### Actual result

User profile can't be saved (in both Front or Back) with an error:

`Save failed with the following error: Duplicate entry '12' for key 'PRIMARY'`

until field is unpublished (field has the id 12)

### System information (as much as possible)
Joomla 3.9.11
MariaDB 10.4.7
PHP 7.2.21

**Comment by ghost** [comment]
@laoneo can you please comment?

**Comment by laoneo** [comment]
Would be good if somebody can try reproduce the issue, because on a quick test I couldn't on a fresh Joomla installation. Does it only happen for users or is it a general radio field issue?

**Comment by solracsf** [comment]
@laoneo I've only tested with Radio and User as i don't need anything else.

What i don't understand is, why Joomla is trying to rewrite the same original field (id), and not trying to write a new value for that field (id) **AND** user (id)?

**Comment by ghost** [comment]
Can't confirm. Radio custom field is editable and can be saved in Front- and Backend as expected.

#### System Information
Setting | Value
-- | --
PHP Built On | Linux lamp122.cloudaccess.net 3.10.0-962.3.2.lve1.5.24.7.el6h.x86_64 #1 SMP Mon Dec 17 12:02:35 EST 2018 x86_64
Database Type | mysql
Database Version | 5.7.24-cll-lve
Database Collation | utf8mb4_general_ci
Database Connection Collation | utf8mb4_general_ci
PHP Version | 7.3.8
Web Server | Apache
WebServer to PHP Interface | cgi-fcgi
Joomla! Version | Joomla! 3.9.12-dev Development [ Amani ] 12-August-2019 17:15 GMT
Joomla! Platform Version | Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
User Agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:69.0) Gecko/20100101 Firefox/68.0

**Comment by solracsf** [comment]
Have you both tried with **several** users?
Because in my side, field can be saved with the first user that tries to write it, not the next ones (it produces the DUPLICATE KEY error).

**Comment by ghost** [comment]
> Have you both tried with **several** users?

No, as it was not in "Steps to reproduce the issue".

**Comment by solracsf** [comment]
I've tried it on a fresh Joomla, can't reproduce it too.
Yet, it happens on my >3K users Joomla installation.

> What i don't understand is, why Joomla is trying to rewrite the same original field (id), and not trying to write a new value for that field (id) **AND** user (id)?

To clarify, Joomla is trying to rewrite the `_fields` table instead of writing a new value inside the `_fields_values` table.

Here, this is what i get when i save my custom field with **the first (random) user** i try to save the field.
![image](https://user-images.githubusercontent.com/12234510/64263466-f3ab1980-cf2f-11e9-9297-e83707a451de.png)

But when i try to save it **another (random) user**:
![image](https://user-images.githubusercontent.com/12234510/64263585-235a2180-cf30-11e9-8252-d5a679bd665a.png)

And of course, nothing is write to the `_fields_values` table 😕 

I've tried to truncate all `_fields*` tables and restart creating the field, same result.
